core/src/main/java/org/apache/iceberg/BaseTransaction.java

```diff
@@ -479,6 +480,49 @@ private void applyUpdates(TableOperations underlyingOps) {
           // Cannot pass even with retry due to conflicting metadata changes. So, break the
           // retry-loop.
           throw new PendingUpdateFailedException(e);
+
+        } catch (ValidationException e) {
+          if (PropertyUtil.propertyAsBoolean(
+              base.properties(),
+              TableProperties.ROLLBACK_COMPACTION_ON_CONFLICTS_ENABLED,
+              TableProperties.ROLLBACK_COMPACTION_ON_CONFLICTS_ENABLED_DEFAULT)) {
```

Comment:
thanks for the update! In that case should we call this something like `ROLLBACK_REWRITE_ON_CONFLICTS_ENABLED`?

Reply:
also the term rollback seems to be not accurate. We are not just rolling back, but we are also reapplying the changes after rolling back. Maybe something like `REAPPLY_REPLACE_SNAPSHOT_ON_CONFLICTS_ENABLED`?

Reply:
> We are not just rolling back, but we are also reapplying the changes after rolling back. Maybe something like REAPPLY_REPLACE_ON_CONFLICTS_ENABLED? Any other suggestions for the name?

makes sense, how about something like `RETRY_WITH_ROLLBACK_REPLACE_ON_CONFLICTS_ENABLED`, it gives an impression that we are retrying the applying the changes with roll-backing replace operations snapshot.